File: 백준/Gold/1806. 부분합/부분합.py
import sys

N, S = map(int, sys.stdin.readline()[:-1].split())
nums = list(map(int, sys.stdin.readline()[:-1].split()))

left, right = 0, 0 # 투 포인터
s = 0
length = int(10e9) # 가장 작은 길이

while True:
    if s >= S:
        length = min(length, right - left); s -= nums[left]
        left += 1
    elif right == N: break
    else:
        s += nums[right]
        right += 1

if length == int(10e9): print(0)
else: print(length)

File: 백준/Gold/1806. 부분합/부분합.cc
#include <iostream>
#include <vector>
#include <limits.h>
using namespace std;

#define INF INT_MAX
#define MAX 100000

int N, S;
int nums[MAX];

int main() {
    cin >> N;
    cin >> S;

    for (int i=0; i<N; i++){
        cin >> nums[i];
    }

    int left = 0;
    int right = 0;
    int sum = 0;
    int length = INT_MAX;

    while (1){
        if (sum >= S){
            length = min(length, right - left);
            sum -= nums[left];
            left++;
        }
        else if (right == N){
            break;
        }
        else{
            sum += nums[right];
            right++;
        }
    }

    if (length == INT_MAX){
        cout << 0;
    }
    else{
        cout << length;
    }

    return 0;
}